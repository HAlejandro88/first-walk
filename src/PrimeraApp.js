import React, { Fragment } from 'react'; //imr
import PropTypes from 'prop-types';


const PrimeraApp = ({ saludo, subtitulo }) => {// Fragment // destructurar la propiedades

    
    return (
        <>
            <h1>{ saludo }</h1>
            <p>Mi primera aplicacion</p>
            <h4>{ subtitulo }</h4>
        </>
    );
}

PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired // si pones el isRequired el valor debe de ir 
}


PrimeraApp.defaultProps = {
    subtitulo: 'propiedad por default'
}

export default PrimeraApp;