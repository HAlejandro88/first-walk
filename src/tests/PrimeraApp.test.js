import React from 'react';
import PrimeraApp from '../PrimeraApp';
/* import { render } from '@testing-library/react'; */
import '@testing-library/jest-dom';
import { shallow } from 'enzyme'; // todo el dom 




describe('pruebas en la <PrimeraApp>', () => {
    /* test('demostrar le mensaje Hola soy alex', () => {
        const saludo = 'Hola soy alex';
        const { getByText }  = render(<PrimeraApp saludo={ saludo }/>);
        expect(getByText(saludo)).toBeInTheDocument();
    }); */

    test('debe de mostrar el compomnente <PrimeraApp> correctaemnte', () => {
        const saludo = 'Hola soy alex';
        const wrapper = shallow(<PrimeraApp saludo={saludo} />);
        expect(wrapper).toMatchSnapshot();
    });

    /* test('debe de mostrar el subtitulo enviado por props', () => {
        const saludo = 'Hola soy alex';
        const subTitulo = 'propiedad por default';

        const wrapper = shallow(<PrimeraApp saludo={saludo} subtitulo={ subTitulo } />);
        
        const textoParrafo = wrapper.find('p').text();
        expect( textoParrafo ).toBe(subTitulo);
    }); */
});