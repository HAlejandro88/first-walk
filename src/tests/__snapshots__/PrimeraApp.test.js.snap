// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en la <PrimeraApp> debe de mostrar el compomnente <PrimeraApp> correctaemnte 1`] = `
<Fragment>
  <h1>
    Hola soy alex
  </h1>
  <p>
    Mi primera aplicacion
  </p>
  <h4>
    propiedad por default
  </h4>
</Fragment>
`;
